echopype/echodata/combine.py

```diff
@@ -435,14 +446,296 @@ def _check_echodata_channels(
+        List of Datasets to be combined
+    ed_group: str
+        The name of the ``EchoData`` group being combined
+
+    Returns
+    -------
+    None
+
+
+    Raises
+    ------
+    RuntimeError
+        If the timeX dimension is not in ascending order
+        for the specified echodata group
+    """
+
+    # get all time dimensions of the input Datasets
+    ed_time_dim = set(ds_list[0].dims).intersection(POSSIBLE_TIME_DIMS)
+
+    for time in ed_time_dim:
+        # gather the first time of each Dataset
+        first_times = []
+        for ds in ds_list:
+            times = ds[time].values
+            if isinstance(times, np.ndarray):
+                # store first time if we have an array
+                first_times.append(times[0])
+            else:
+                # store first time if we have a single value
+                first_times.append(times)
+
+        first_times = np.array(first_times)
+
+        # skip check if all first times are NaT
+        if not np.isnan(first_times).all():
+            is_descending = (np.diff(first_times) < np.timedelta64(0, "ns")).any()
+
+            if is_descending:
+                raise RuntimeError(
+                    f"The coordinate {time} is not in ascending order for "
+                    f"group {ed_group}, combine cannot be used!"
+                )
+
+
+def _check_filter_params(
+    ds_list: List[xr.Dataset], ed_group: Literal["Vendor_specific"], ds_append_dims: set
+) -> None:
+    """
+    Check for identical filter params for all inputs in Vendor specific group
+
+    Parameters
+    ----------
+    ds_list: list of xr.Dataset
+        List of Datasets to be combined
+    ed_group: "Vendor_specific"
+        The name of the ``EchoData`` group being combined,
+        this is only works for "Vendor_specific" group.
+    ds_append_dims: set
+        A set of datasets append dimensions
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    ValueError
+        If ``ed_group`` is not ``Vendor_specific``.
+    RuntimeError
+        If non identical filter parameters is found.
+    """
+    if ed_group != "Vendor_specific":
+        raise ValueError("Group must be `Vendor_specific`!")
+
+    if len(ds_append_dims) > 0:
+        # If there's a dataset appending dimension, drop for comparison
+        # of the other values... everything else should be identical
+        ds_list = [ds.drop_dims(ds_append_dims) for ds in ds_list]
+
+    it = iter(ds_list)
+    # Init as identical, must stay True.
+    is_identical = True
+    dataset = next(it)
+    for next_dataset in it:
+        is_identical = dataset.identical(next_dataset)
+        if not is_identical:
+            raise RuntimeError(
+                f"Non identical filter parameters in {ed_group} group. " "Objects cannot be merged!"
+            )
+        dataset = next_dataset
+
+
+def _merge_attributes(attributes: List[Dict[str, str]]) -> Dict[str, str]:
+    """
+    Merges a list of attributes dictionary
```

Comment:
super minor: just to keep consistency with other docstring
```suggestion
    Merge a list of attributes dictionary
```